src/View/Helper/PaginatorHelper.php

```diff
@@ -1075,8 +1075,9 @@ public function first($first = '<< first', array $options = [])
             }
         } elseif ($params['page'] > 1 && is_string($first)) {
             $first = $options['escape'] ? h($first) : $first;
+            $url = array_merge($options['url'], ['page' => 1]);
```

Comment:
Interestingly, this means you can control the page options, but I don't think that's going to cause any problems.

Reply:
Wouldn't the merge overwrite anything in options[url]?

Reply:
It won't overwrite 'page' or 'pageCount'. Should we pass $options['url'] as the 3rd parameter to generateUrl() like your other PR?

Reply:
The signature of generateUrl() is different in 3.x and 4.x. In 4.x there is an addtional url parameter that is being used in the PR you referenced. I can't make a similar change here as it would require changing the public interface.